src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -874,6 +846,13 @@ private TypedConstant GetDefaultValueArgument(ParameterSymbol parameter, Attribu
             {
                 return new TypedConstant(parameterType, kind, defaultValue);
             }
+
+            static int getCallerArgumentArgumentIndex(ParameterSymbol parameter, ImmutableArray<int> argumentsToParams)
+            {
+                return argumentsToParams.IsDefault
+                        ? parameter.CallerArgumentExpressionParameterIndex
+                        : argumentsToParams.IndexOf(parameter.CallerArgumentExpressionParameterIndex);
```

Comment:
>parameter.CallerArgumentExpressionParameterIndex [](http://example.com/codeflow?start=52&length=48)

Suppose ```parameter.CallerArgumentExpressionParameterIndex``` is -1 and there is an argument that doesn't map to any parameter (a name doesn't match or an extra argument is provided). What is going to happen then? We probably should add test(s) like that.

Reply:
@AlekseyTs I'm not sure what exactly we want to test here. Extra argument or unmatched named argument would produce an error elsewhere, so I assume such a test shouldn't be affected by the logic here? Please correct me. I'm certainly open to add more tests, but I'm not getting which scenario I should try to hit in the test.

Reply:
>  I'm not sure what exactly we want to test here. Extra argument or unmatched named argument would produce an error elsewhere, so I assume such a test shouldn't be affected by the logic here? Please correct me. I'm certainly open to add more tests, but I'm not getting which scenario I should try to hit in the test.

I'll try to explain why I am asking for more testing here. The logic here as it stands right now feels fragile to me. In particular, if ```parameter.CallerArgumentExpressionParameterIndex``` is -1, we want the enclosing local function to return -1. However, we are unconditionally (without checking the value of ```parameter.CallerArgumentExpressionParameterIndex```) trying to look it up in the array. It is quite possible that this code is unreachable with ` -1`s in the array, therefore, the lookup would always return -1 if we are looking for -1. This, however, depends on things that we don't control here locally. Therfore I am asking for tests that would demonstrate the "unreachability" of the code under the specified conditions. I made some suggestions, but these are just gueses at this point and likely there are other scenarios to try out. My ultimate goal is to confirm that this local function never returns something other than -1 when it is called with ```parameter.CallerArgumentExpressionParameterIndex```  -1. If I was writing the code, I would take an alternative approach. I would explicitly check if ```parameter.CallerArgumentExpressionParameterIndex```  is -1 before doing the lookup. This would make the implementation more robust and independent from what other components do.

Reply:
@AlekseyTs Thanks a ton! Yes I wrote the code under the assumption that the array won't contain -1.
I explicitly checked for `CallerArgumentExpressionParameterIndex == -1` in the last commit. Does that do job?